app/compare/diff_preprocessing.tsx

```diff
@@ -0,0 +1,161 @@
+import { invocation as invocation_proto } from "../../proto/invocation_ts_proto";
+import { command_line } from "../../proto/command_line_ts_proto";
+
+/**
+ * A unique token used to identify parts of the diff that were hidden due to pre-processing options.
+ */
+export const HIDDEN_TOKEN = `/__HIDDEN__2540ee7b-c0b0-4ed6-ab90-b68c220cc9ea/`;
+/**
+ * protobuf.js' JSON representation of HIDDEN_TOKEN when stored in a `bytes` field.
+ */
+const HIDDEN_TOKEN_BYTES_ENCODING = "AAAAAAAAAAAAAAACBQQAAAAHAAAAAAAAAAQAAAYAAAAJAAAABggAAgIAAAAJAAAA";
```

Comment:
this is getting pretty gnarly lol

what if instead of replacing fields with HIDDEN_TOKEN, we just removed them the object altogether (either set them to `undefined` or used `delete invocation.foo`

that way we wouldn't have to do any of this hidden token stuff,  we'd get rid of LineContent, and i bet we could get rid of a bunch of the complicated logic below / simplify things

Reply:
That works, but I feel like it's a better user experience if we are explicit about the changes that we're making.

If we just delete stuff rather than showing it as hidden, we could accidentally delete something and never know that we deleted it.

This also gives the user explicit feedback that they are not seeing something because of a setting (which is applied by default -- and that they have the option to see it if they are really interested).

Reply:
i'm weary of adding a lot of complex code that we have to maintain for a minor usability improvement (in this case slightly grayer "(hidden)" text)

my big thing here is not having a hardcoded `AAAAAAAAAAAAAAACBQQAAAAHAAAAAAAAAAQAAAYAAAAJAAAABggAAgIAAAAJAAAA` string. i also don't really like `/__HIDDEN__2540ee7b-c0b0-4ed6-ab90-b68c220cc9ea/`. thought we could simplify the rest of this code by just making a reasonable design decision (like just hiding "hidden" fields), but as long as we get rid of those i can probably live with this

how do you generate a string like that if you ever wanted to change this code?

i also don't like that when you want to redact a new field (which will probably happen a fair number of times as we get new builds with new fields we haven't seen before), you have to figure out where to do it. 

do you modify the invocation proto? do you write a regex to clear it out of the json? do you shove one thing into a proto, and then regex another thing out of the json? the decisions we make here cost us engineering time in the future - which with a small team is the most valuable resource we have

Reply:
OK - I was originally not optimizing for maintenance cost here since I figured this would be mostly throwaway code. But I guess if we want to give this a permanent home in the "raw" tab it makes sense to put in more effort to making this cleaner. Done